static/src/javascripts/projects/commercial/modules/dfp/create-slots.ts

```diff
@@ -194,61 +193,103 @@ const createAdSlotElements = (
+ *
+ */
+const mergeSizeMappings = (
+	defaultSizeMappings: SizeMappings,
+	optionSizeMappings: CreateSlotOptions['sizes'],
+): SizeMappings => {
+	if (!optionSizeMappings) return defaultSizeMappings;
+	const mergedSizeMappings: SizeMappings = { ...defaultSizeMappings };
+	const optionDevices = Object.keys(optionSizeMappings);
+	for (let i = 0; i < optionDevices.length; i++) {
+		const device = optionDevices[i];
+		const optionSizeMappingsForDevice = optionSizeMappings[device];
+		const defaultSizeMappingsForDevice = defaultSizeMappings[device];
+		// eslint-disable-next-line @typescript-eslint/no-unnecessary-condition -- update tsconfig: "noUncheckedIndexedAccess": true
+		if (defaultSizeMappingsForDevice && optionSizeMappingsForDevice) {
+			mergedSizeMappings[device] = mergedSizeMappings[device].concat(
+				optionSizeMappingsForDevice,
+			);
+		}
+	}
+	return mergedSizeMappings;
 };
 
-export const createSlots = (
+/**
+ * Convert size mappings to a string that will be added to the ad slot
+ * data atttributes.
+ *
+ * e.g. { desktop: [[320,250], [320, 280]] } => { desktop: '320,250|320,280' }
+ *
+ */
+const covertSizeMappingsToStrings = (
+	sizeMappings: SizeMappings,
+): Record<string, string> =>
+	Object.entries(sizeMappings).reduce(
+		(result: Record<string, string>, [device, sizes]) => {
+			result[device] = sizes.join('|');
+			return result;
+		},
+		{},
+	);
+
+/**
+ * Prefix all object keys with data-${key}
+ */
+const createDataAttributes = (
+	attrs: Record<string, string>,
+): Record<string, string> =>
+	Object.entries(attrs).reduce(
+		(result: Record<string, string>, [key, value]) => {
+			result[`data-${key}`] = value;
+			return result;
+		},
+		{},
+	);
+
+export const createAdSlot = (
 	type: string,
 	options: CreateSlotOptions = {},
-): HTMLElement[] => {
-	const attributes: Record<string, string> = {};
-	const definition: AdSlotDefinition = adSlotDefinitions[type];
-	const slotName: string = options.name ?? definition.name ?? type;
-	const classes: string[] =
-		options.classes?.split(' ').map((cn) => `ad-slot--${cn}`) ?? [];
-
-	const sizes: SizeMappings = { ...definition.sizeMappings };
-	const sizeStrings: Record<string, string> = {};
-
-	const optionSizes = options.sizes;
-
-	if (optionSizes) {
-		Object.keys(optionSizes).forEach((optionSize: string) => {
-			const optionSizesArray = optionSizes[optionSize];
-			if (optionSizesArray) {
-				// eslint-disable-next-line @typescript-eslint/no-unnecessary-condition -- update tsconfig: "noUncheckedIndexedAccess": true
-				sizes[optionSize] = sizes[optionSize]
-					? sizes[optionSize].concat(optionSizesArray)
-					: optionSizesArray;
-			}
-		});
-	}
+): HTMLElement => {
+	const adSlotConfig = adSlotConfigs[type];
+	const slotName = options.name ?? adSlotConfig.name ?? type;
 
-	Object.keys(sizes).forEach((size) => {
-		sizeStrings[size] = sizes[size].join('|');
-	});
+	const sizeMappings = mergeSizeMappings(
+		adSlotConfigs[type].sizeMappings,
```

Comment:
Since you've already accessed `adSlotConfigs[type]` on L272 should it be this?
```suggestion
		adSlotConfig.sizeMappings,
```